lib/std/hash/crc.zig

```diff
@@ -1,285 +1,911 @@
-        }
-
-        pub fn hash(input: []const u8) u32 {
-            var c = Self.init();
-            c.update(input);
-            return c.final();
-        }
-    };
-}
-
-const verify = @import("verify.zig");
-
-test "crc32 ieee" {
-    const Crc32Ieee = Crc32WithPoly(.IEEE);
-
-    try testing.expect(Crc32Ieee.hash("") == 0x00000000);
-    try testing.expect(Crc32Ieee.hash("a") == 0xe8b7be43);
-    try testing.expect(Crc32Ieee.hash("abc") == 0x352441c2);
-}
-
-test "crc32 castagnoli" {
-    const Crc32Castagnoli = Crc32WithPoly(.Castagnoli);
-
-    try testing.expect(Crc32Castagnoli.hash("") == 0x00000000);
-    try testing.expect(Crc32Castagnoli.hash("a") == 0xc1d04330);
-    try testing.expect(Crc32Castagnoli.hash("abc") == 0x364b3fb7);
-}
-
-test "crc32 iterative" {
-    try verify.iterativeApi(Crc32WithPoly(.IEEE));
-}
-
-// half-byte lookup table implementation.
-pub fn Crc32SmallWithPoly(comptime poly: Polynomial) type {
-    return struct {
-        const Self = @This();
-        const lookup_table = block: {
-            var table: [16]u32 = undefined;
-
-            for (&table, 0..) |*e, i| {
-                var crc = @as(u32, @intCast(i * 16));
-                var j: usize = 0;
-                while (j < 8) : (j += 1) {
-                    if (crc & 1 == 1) {
-                        crc = (crc >> 1) ^ @intFromEnum(poly);
-                    } else {
-                        crc = (crc >> 1);
-                    }
-                }
-                e.* = crc;
-            }
-
-            break :block table;
-        };
-
-        crc: u32,
-
-        pub fn init() Self {
-            return Self{ .crc = 0xffffffff };
-        }
-
-        pub fn update(self: *Self, input: []const u8) void {
-            for (input) |b| {
-                self.crc = lookup_table[@as(u4, @truncate(self.crc ^ (b >> 0)))] ^ (self.crc >> 4);
-                self.crc = lookup_table[@as(u4, @truncate(self.crc ^ (b >> 4)))] ^ (self.crc >> 4);
-            }
-        }
-
-        pub fn final(self: *Self) u32 {
-            return ~self.crc;
-        }
-
-        pub fn hash(input: []const u8) u32 {
-            var c = Self.init();
-            c.update(input);
-            return c.final();
-        }
-    };
-}
-
-test "small crc32 iterative" {
-    try verify.iterativeApi(Crc32SmallWithPoly(.IEEE));
+test {
+    _ = @import("crc/test.zig");
 }
 
-test "small crc32 ieee" {
-    const Crc32Ieee = Crc32SmallWithPoly(.IEEE);
-
-    try testing.expect(Crc32Ieee.hash("") == 0x00000000);
-    try testing.expect(Crc32Ieee.hash("a") == 0xe8b7be43);
-    try testing.expect(Crc32Ieee.hash("abc") == 0x352441c2);
-}
-
-test "small crc32 castagnoli" {
-    const Crc32Castagnoli = Crc32SmallWithPoly(.Castagnoli);
-
-    try testing.expect(Crc32Castagnoli.hash("") == 0x00000000);
-    try testing.expect(Crc32Castagnoli.hash("a") == 0xc1d04330);
-    try testing.expect(Crc32Castagnoli.hash("abc") == 0x364b3fb7);
-}
+pub const Crc3Gsm = Crc(u3, .{
```

Comment:
[same problem](https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces)

```diff
-std.hash.crc.Crc3Gsm
+std.hash.crc.@"3Gsm"
```

or perhaps `std.hash.crc.catalog.@"3GSm"`